platform/darwin/src/NSExpression+MGLAdditions.h

```diff
@@ -9,6 +9,128 @@
+ This attribute corresponds to the `cubic-bezier` value in the
+ <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-interpolate"><code>interpolate</code></a>
+ expression reference in the Mapbox Style Specification.
+ */
+extern MGL_EXPORT const MGLExpressionInterpolationMode MGLExpressionInterpolationModeCubicBezier;
+
+@interface NSExpression (MGLVariableAdditions)
+
+/**
+ `NSExpression` attribute that corresponds to the
+ <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-zoom"><code>zoom</code></a>
+ expression reference in the Mapbox Style Specification.
+ */
++ (NSExpression *)mgl_zoomLevelVariableExpression;
+
+/**
+ `NSExpression` attribute that corresponds to the
+ <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-heatmap-density"><code>heatmap-density</code></a>
+ expression reference in the Mapbox Style Specification.
+ */
++ (NSExpression *)mgl_heatmapDensityVariableExpression;
+
+/**
+ `NSExpression` attribute that corresponds to the
+ <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#eexpressions-geometry-type"><code>geometry-type</code></a>
+ expression reference in the Mapbox Style Specification.
+ */
++ (NSExpression *)mgl_geometryTypeVariableExpression;
+
+/**
+ `NSExpression` attribute that corresponds to the
+ <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-id"><code>id</code></a>
+ expression reference in the Mapbox Style Specification.
+ */
++ (NSExpression *)mgl_featureIdentifierVariableExpression;
+
+/**
+ `NSExpression` attribute that corresponds to the
+ <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-properties"><code>properties</code></a>
+ expression reference in the Mapbox Style Specification.
+ */
++ (NSExpression *)mgl_featurePropertiesVariableExpression;
+
+@end
+
+@interface NSExpression (MGLInitializerAdditions)
+
+/**
+ Returns a conditional function expression specifying the string predicate, and
+ expressions for each condition.
+ 
+ @param conditionPredicate The predicate to get evaluated.
+ @param trueExpression The expression for conditions equal to true.
+ @param falseExpression The expression for conditions equal to false.
+ */
++ (instancetype)mgl_expressionForConditional:(nonnull NSPredicate *)conditionPredicate trueExpression:(nonnull NSExpression *)trueExpression falseExpresssion:(nonnull NSExpression *)falseExpression NS_SWIFT_NAME(mgl_conditional(_:trueExpression:falseExpression:));
+
+/**
+ Returns a step function expression specifying the function operator, default expression
+ and stops.
+ 
+ @param input The input expression.
+ @param from The expression which could be a constant or function expression.
+ @param stops The stops dictionay must be numeric literals in strictly ascending order.
+ */
++ (instancetype)mgl_expressionForStepFunction:(nonnull NSExpression*)input from:(nonnull NSExpression *)from stops:(nonnull NSExpression*)stops NS_SWIFT_NAME(mgl_stepFunction(input:from:stops:));
```

Comment:
“Function” here is ambiguous as to whether it refers to an expression function or a style function, which could be problematic for developers migrating from style functions. I wonder if we arrived at this name in order to align with `+expressionForFunction:arguments:`. The difference is that “function” in that method describes the parameter, which is the name of a function, whereas here it has nothing to do with the first parameter. If we wanted to align with that method, we’d have to call it `+mgl_stepExpressionForInputExpression:minimumExpression:stops:`.

The aftermarket expression function is called `mgl_step:from:stops:`. This initializer should be called either `+mgl_expressionForSteppingExpression:minimum:stops:` or `+mgl_expressionForSteppingExpression:fromExpression:stops:`. There may need to be a Swift name for this initializer as well, to avoid the problem in #11612.